File: Java/rango.java
import java.util.Scanner;

public class rango {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Ingresa un número:");

        String input = sc.nextLine();
        try {
            double val1 = Double.parseDouble(input);
            if (val1 >= 0 && val1 <= 10) {
                System.out.println("El valor está entre 0 y 10");
            } else {
                System.out.println("El valor está fuera del rango de 0 y 10");
            }
        } catch (NumberFormatException e) {
            System.out.println("Entrada inválida: no es un número.");
        }

        sc.close();
    }
}

File: JavaScript/rango.js
let a;

do {
    try {
        a = parseFloat(prompt("Ingresa un número entre 0 y 10: "));
        
        if (isNaN(a)) {
            throw new Error("El valor ingresado no es un número válido.");
        }
        
        if (a >= 0 && a <= 10) {
            console.log("El número está dentro del rango");
            break;
        } else {
            console.log("El valor está fuera del rango");
        }
    } catch (error) {
        console.log(error.message);
    }
} while (true);
